File: Backend_Ecommerce/src/main/java/com/commerce/dto/UserPrincipal.java
package com.commerce.dto;

import com.commerce.datamodel.User;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UserPrincipal
{
  private final int id;
  private final String username;
  private final Instant loginTime;
  
  public UserPrincipal(User user)
  {
    this.id = user.getId();
    this.username = user.getUsername();
    this.loginTime = Instant.now();
  
  }
  
  
  public String subject()
  {
    return String.format("%s,%s,%s", id, username, loginTime);
  }
  
  public static UserPrincipal fromSubject(String subject)
  {
    String[] split = subject.split(",");
    return new UserPrincipal((int) Long.parseLong(split[0]), split[1], Instant.parse(split[2]));
  }
  
  
}

File: C# Backend/E-commerce/E-commerce/DTO/UserPrincipal.cs
using E_commerce.DataModel;
using Microsoft.Identity.Client;

namespace E_commerce.DTO
{
    public class UserPrincipal
    {
        private IAccount account;

        public int Id { get; }
        public string Username { get; }
        public DateTime LoginTime { get; }


        public UserPrincipal(int id, string username, DateTime loginTime)
        {
            Id = id;
            Username = username;
            LoginTime = loginTime;
        }

        public UserPrincipal(User user)
        {
            Id = user.Id;
            Username = user.Username;
            LoginTime = DateTime.Now;
        }

        public UserPrincipal(IAccount account)
        {
            this.account = account;
        }

        public string Subject()
        {
            return $"{Id},{Username},{LoginTime}";
        }

        public static UserPrincipal FromSubject(string subject)
        {
            string[] split = subject.Split(",");
            int id = int.Parse(split[0]);
            string username = split[1];
            DateTime loginTime = DateTime.Parse(split[2]);
            return new UserPrincipal(id, username, loginTime);
        }
    }
}
